packages/charts/react-charting/src/components/DonutChart/Arc/Arc.tsx

```diff
@@ -31,12 +31,11 @@ export class Arc extends React.Component<IArcProps, IArcState> {
   }
 
   public render(): JSX.Element {
-    const { arc, href, focusedArcId } = this.props;
+    const { arc, href, focusedArcId, activeArc } = this.props;
     const getClassNames = classNamesFunction<IArcStyleProps, IArcStyles>();
     const id = this.props.uniqText! + this.props.data!.data.legend!.replace(/\s+/, '') + this.props.data!.data.data;
-    const opacity: number =
-      this.props.activeArc === this.props.data!.data.legend || this.props.activeArc === '' ? 1 : 0.1;
-
+    const isActive = activeArc?.includes(this.props.data?.data.legend!);
```

Comment:
>const isActive  [](http://example.com/codeflow?start=4&length=15)

why repeating this logic.

Reply:
so that it can be used to calculate both opacity and data-is-focusable without the need to add same checks again